docs/ADR-29-refactor-hud-controller.md

```diff
@@ -0,0 +1,77 @@
+# ADR-29 - Refactor HUD control
+
+## Abstract
+
+Currently, the Kernel controls the UI in Unity, with a component inside Unity named `HUDController`. It has many responsibilities breaking the single responsibility principle, making it challenging to adapt and change. Besides that, the architecture doesn't take into account a way of making the HUDs variables for different platforms.
+
+To deal with those problems, we're going to split those responsibilities and modify the architecture of the `HUDController` to be extendable to different platform implementations, like desktop or mobile, and port the control of the HUD from Kernel to Unity progressively to make Kernel smaller and less responsible.
+
+## Needs
+
+* Remove unnecessary responsibilities from Kernel.
+* Split `HUDController` responsibilities.
+* Make a `HUD System` extendable to different platform implementations, like desktop or mobile.
+
+## Current implementation
+
+### Kernel and Renderer communication
+We have multiple HUDs synced between Kernel and Renderer.
+
+When we start Decentraland Explorer, the Kernel will send a message to Unity to pre-load and show some HUDs that they live in `InitialScene` with a `HUDController` script. Then, the Kernel can hide or show those HUDs as it needs.
+
+![resources/ADR-29/diagram-1.svg](resources/ADR-29/diagram-1.svg)
+
+### HUDController architecture
+
+`HUDController` is a `MonoBehavior` that lives in the `InitialScene`. It is responsible for creating HUDs, keeping each HUD's lifecycle, reacting to the Kernel messages, and different interconnecting HUDs.
+
+### Approach
+
+So keeping in mind the objectives, we need to create a way to refactor the `HUD System` progressively. Thus, we will divide it into three stages.
+
+#### First stage
+
+This stage's idea is to split the responsibility of instantiation in the `HUDController` and make it extendable.
+
+So, to accomplish this goal, the `abstract factory design pattern` will be implemented.
+
+The classes will look like the following:
```

Comment:
For some reason, the github preview put all these items under the same line. We should add line breaks or itemization here.